daemon/daemon.go

```diff
@@ -207,68 +216,88 @@ func (daemon *Daemon) restore() error {
 		return err
 	}
 
+	loadWg := syncext.NewLimitedWaitGroup(parallelStartupJobs)
```

Comment:
If you don't mind, I'd, personally, prefer not to add the new package for this limited case, especially since I feel like it's weird to mix the two things.

Reply:
Would you prefer that I use `WaitGroup` and `chan struct{}` separately, or that I make `LimitedWaitGroup` a new type *just* for `daemon/daemon.go`?

Reply:
Separately 🙏👼

Reply:
Sure, though note that this will change

```go
	loadWg := syncext.NewLimitedWaitGroup(parallelStartupJobs)
	for _, v := range dir {
		loadWg.Add()
		go func(id string) {
			loadWg.Up()
			defer loadWg.Down()
			// ...
		}(v.Name())
	}
	loadWg.Wait()
```

to

```go
	var loadWg sync.WaitGroup
	loadSem := make(chan struct{}, parallelStartupJobs)
	for _, v := range dir {
		loadWg.Add(1)
		go func(id string) {
			loadSem <- struct{}{}
			defer func() {
				<-loadSem
				loadWg.Done()
			}()
			// ...
		}(v.Name())
	}
	loadWg.Wait()
```

Which is the main reason I combined them (the latter seems far easier to mess up on copy-paste and would be repeated quite a bit). But sure, I can do the conversion if you prefer.